Windows builds with updated msys/mingw gcc segfault while building skia

```
mkdir -p `dirname /c/projects/servo/target/debug/build/servo-skia-d260e1da355c4a73/out/src/core/SkDistanceFieldGen.o` && g++ -c -O3 -DSK_RELEASE -DGR_RELEASE=1 -DSK_A32_SHIFT=24 -DSK_R32_SHIFT=16 -DSK_G32_SHIFT=8 -DSK_B32_SHIFT=0 -DGR_GL_USE_NEW_SHADER_SOURCE_SIGNATURE=1 -I"/include" -I"/include" -Iinclude/config -Iinclude/core -Iinclude/effects -Iinclude/gpu -Iinclude/images -Iinclude/pathops -Iinclude/pipe -Iinclude/ports -Iinclude/utils -Isrc/core -Isrc/image -Isrc/opts -Isrc/ports -Isrc/sfnt -Isrc/utils -Ithird_party/etc1 -Ithird_party/ktx -iquote src/gpu -I/usr/include/freetype2 -std=gnu++0x -Wno-deprecated-declarations -D_WIN32_WINNT=0x0600 -DSK_BUILD_FOR_WIN32 -DSK_INCLUDE_FREETYPE -Iinclude/utils/win -I/mingw64/include/freetype2 -msse2 -mfpmath=sse -o /c/projects/servo/target/debug/build/servo-skia-d260e1da355c4a73/out/src/core/SkDistanceFieldGen.o src/core/SkDistanceFieldGen.cpp
--- stderr
src/core/SkDistanceFieldGen.cpp: In function 'bool generate_distance_field_from_image(unsigned char*, const unsigned char*, int, int)':
src/core/SkDistanceFieldGen.cpp:332:13: internal compiler error: Segmentation fault
 static bool generate_distance_field_from_image(unsigned char* distanceField,
             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
libbacktrace could not find executable to open
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://sourceforge.net/projects/msys2> for instructions.
make: *** [makefile.cargo:776: /c/projects/servo/target/debug/build/servo-skia-d260e1da355c4a73/out/src/core/SkDistanceFieldGen.o] Error 1
make: *** Waiting for unfinished jobs....
thread 'main' panicked at 'assertion failed: Command::new("make").args(&["-R", "-f", "makefile.cargo",
                            &format!("-j{}" , env :: var ( "NUM_JOBS" ) .
                                     unwrap (
                                     ))]).status().unwrap().success()', C:/projects/servo/.cargo\registry\src\github.com-1ecc6299db9ec823\servo-skia-0.20130412.10\build.rs:12
note: Run with `RUST_BACKTRACE=1` for a backtrace.
cargo build -v
Build completed in 0:25:16
Command exited with code 101
```

--- a/README.md
+++ b/README.md
@@ -80,7 +80,9 @@ Download Python for Windows [here](https://www.python.org/downloads/release/pyth
 required for the SpiderMonkey build on Windows.
 
 Install MSYS2 from [here](https://msys2.github.io/). After you have done so, open an MSYS shell
-window and update the core libraries and install new packages:
+window and update the core libraries and install new packages. The extra step at the end is to
+downgrate GCC to 5.4, as the GCC6 versions in mingw currently fail to compile some of our
+dependencies. We are upgrading to a gcc-free build on Windows as soon as possible:
 
 ```sh
 pacman -Su
@@ -88,6 +90,11 @@ pacman -Sy git mingw-w64-x86_64-toolchain mingw-w64-x86_64-freetype \
     mingw-w64-x86_64-icu mingw-w64-x86_64-nspr mingw-w64-x86_64-ca-certificates \
     mingw-w64-x86_64-expat mingw-w64-x86_64-cmake tar diffutils patch \
     patchutils make python2-setuptools
+export GCC_URL=http://repo.msys2.org/mingw/x86_64/mingw-w64-x86_64-gcc
+export GCC_EXT=5.4.0-1-any.pkg.tar.xz
+pacman -U --noconfirm $GCC_URL-$GCC_EXT $GCC_URL-ada-$GCC_EXT \
+    $GCC_URL-fortran-$GCC_EXT $GCC_URL-libgfortran-$GCC_EXT $GCC_URL-libs-$GCC_EXT \
+    $GCC_URL-objc-$GCC_EXT
 easy_install-2.7 pip virtualenv
 ```
 

